tests/suites/test_suite_x509parse.function

```diff
@@ -302,9 +302,10 @@ int verify_parse_san( mbedtls_x509_subject_alternative_name *san,
     return( 0 );
 }
 
-int parse_crt_ext_cb( mbedtls_x509_crt const *crt, mbedtls_x509_buf const *oid, int critical,
-                      const unsigned char *p, const unsigned char *end )
+int parse_crt_ext_cb( void *p_ctx, mbedtls_x509_crt const *crt, mbedtls_x509_buf const *oid,
+                      int critical, const unsigned char *p, const unsigned char *end )
 {
+    ( void ) p_ctx;
```

Comment:
will do but why is this a requirement when ca_callback_fail in the same file doesn't have a check on the void *data pointer?

Reply:
I don't know why it wasn't done for `ca_callback_fail`, I wasn't an author or a reviewer of that. We don't have precise metrics for test coverage (well, except for a line coverage metric which is not very meaningful because we only compute it in one particular configuration), but generally it's a good idea to have complete branch coverage except for “can't happen” cases, and complete data path coverage. The library has a lot of technical debt around test coverage, but that's no reason not to do things properly in new code.

Reply:
Implemented by passing the OID of the unsupported extension to be parsed to the callback using the pointer, so that if the library passes the wrong pointer the test fails.

See e58b4638e594a0ac16a19e40d96da09508f0a7ff